### ISSUE TITLE
Warning with PyTorch 1.4

### ISSUE THREAD

**Issue by djstrong** [facebookresearch/adaptive-span#14]
UserWarning: Detected call of `lr_scheduler.step()` before `optimizer.step()`. In PyTorch 1.1.0 and later, you should call them in the opposite order: `optimizer.step()` before `lr_scheduler.step()`.  Failure to do this will result in PyTorch skipping the first value of the learning rate schedule. See more details at https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate
  "https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate", UserWarning)

**Comment by prajjwal1** [comment]
You can send a PR for this.

**Comment by djstrong** [comment]
I changed it in https://github.com/facebookresearch/adaptive-span/blob/d882404be50f488d85683b8b925f0c6aef33e9f3/trainer.py#L67 but `scheduler.step` is also called in `_load_checkpoint`.

**Comment by tesatory** [comment]
Yes, I've seen this warning, but don't think it matters that much since our warming up takes thousands of steps and shift by one step should be negligible.

**Comment by djstrong** [comment]
Great, thanks for explanation.
